sdk/core/core-util/src/aborterUtils.ts

```diff
@@ -61,3 +71,23 @@ export function createAbortablePromise<T>(
     abortSignal?.addEventListener("abort", onAbort);
   });
 }
+
+/**
+ * promise.race() implementation that aborts losers as soon as the first promise fulfills.
+ */
+export async function racePromisesAndAbortLosers<T>(
+  promises: ((abortOptions: AbortOptions) => Promise<T>)[],
+  abortSignal?: AbortSignalLike
+): Promise<T> {
+  const loserAborter = new AbortController();
+  const loserAbortListener = () => {
```

Comment:
nitpick: Listener doesn't seem like the best name because this doesn't listen on anything, it just aborts. Perhaps name it `abort`? We can also call it directly in the finally body down there.